### PR TITLE
feat: bump Go to 1.21.0

### PR DESCRIPTION
**Pull request by odsod**
Ensures compatibility with 1.21 source code.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -4,14 +4,14 @@ updates:
...
     directory: /
     schedule:
-      interval: weekly
+      interval: monthly
 
   - package-ecosystem: gomod
     directory: .sage
     schedule:
-      interval: weekly
+      interval: monthly
```
Comment (tvkn) [inline .github/dependabot.yml:17]: Ok to scope creep that in a Go version bump? :)

File: .github/codecov.yml [hunk 2]
```diff
@@ -1,9 +1,6 @@
 codecov:
   require_ci_to_pass: yes
 
-ignore:
-  - internal/examples/proto/gen
```
Comment (tvkn) [inline .github/codecov.yml:5]: Ok to scope creep that in a Go version bump? :)

**Review by tvkn — APPROVED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (odsod) [reply .github/dependabot.yml:17]: :see_no_evil:

File: .github/codecov.yml (hunk 2 above)
Comment (odsod) [reply .github/codecov.yml:5]: :see_no_evil:
